src/backend/distributed/shardsplit/shardsplit_logical_replication.c

```diff
@@ -0,0 +1,734 @@
+/*-------------------------------------------------------------------------
+ *
+ * shardsplit_logical_replication.c
+ *
+ * Function definitions for logically replicating split children.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "miscadmin.h"
+#include "nodes/pg_list.h"
+#include "distributed/colocation_utils.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/shardinterval_utils.h"
+#include "distributed/connection_management.h"
+#include "distributed/remote_commands.h"
+#include "distributed/shard_split.h"
+#include "distributed/listutils.h"
+#include "distributed/shardsplit_logical_replication.h"
+#include "distributed/resource_lock.h"
+#include "utils/builtins.h"
+#include "commands/dbcommands.h"
+
+
+static HTAB *ShardInfoHashMapForPublications = NULL;
+
+/* function declarations */
+static void AddPublishableShardEntryInMap(uint32 targetNodeId,
+										  ShardInterval *shardInterval, bool
+										  isChildShardInterval);
+ShardSplitSubscriberMetadata * CreateShardSplitSubscriberMetadata(Oid tableOwnerId, uint32
+																  nodeId,
+																  List *
+																  replicationSlotInfoList);
+static void CreateShardSplitPublicationForNode(MultiConnection *connection,
+											   List *shardList,
+											   uint32_t publicationForTargetNodeId, Oid
+											   tableOwner);
+static char * ShardSplitPublicationName(uint32_t nodeId, Oid ownerId);
+static void DropAllShardSplitSubscriptions(MultiConnection *cleanupConnection);
+static void DropAllShardSplitPublications(MultiConnection *cleanupConnection);
+static void DropAllShardSplitUsers(MultiConnection *cleanupConnection);
+static void DropAllShardSplitReplicationSlots(MultiConnection *cleanupConnection);
+
+
+/*
+ * CreateShardSplitInfoMapForPublication creates a hashmap that groups
+ * shards for creating publications and subscriptions.
+ *
+ * While creating publications and subscriptions, apart from table owners,
+ * placement of child shard matters too. To further understand this, please see
+ * the following example:
+ *
+ * Shard1(on Worker1) is to be split in Shard2 and Shard3 on Worker2 and Worker3 respectively.
+ * Lets assume the owner to be 'A'. The hashmap groups shard list in the following way.
+ *
+ * Map key
+ * =======	               ------     ------
+ * <Worker2, 'A'> ------> |Shard2|-->|Shard1|
+ *                         ------     ------
+ *
+ *                         ------     ------
+ * <Worker3, 'A'> ------> |Shard3|-->|Shard1|
+ *                         ------     ------
+ * Shard1 is a dummy table that is to be created on Worker2 and Worker3.
```

Comment:
Mapping is 1-1 with Publication, Subs and Replication slot? We should add this global thing in shard_split.c

Reply:
Mapping is 1-1 with Publication, Subs and Replication slot? 

This is very specific to the way we are creating pub/subs for shard split functionality.
So I think its fine to keep it in this file rather than expose it in shard_split.c 

shard_split.c  is driving the workflow and all the helpers required for logical replication of splitting are provided by this file.